gems/aws-sdk-core/lib/aws-sdk-core/telemetry.rb

```diff
@@ -0,0 +1,92 @@
+# frozen_string_literal: true
+
+require_relative 'telemetry/base'
+require_relative 'telemetry/no_op'
+require_relative 'telemetry/otel'
+require_relative 'telemetry/span_kind'
+require_relative 'telemetry/span_status'
+
+module Aws
+  # Observability is the extent to which a system's current state can be
+  # inferred from the data it emits. The data emitted is commonly referred
+  # as Telemetry. The AWS SDK for Ruby currently supports traces as
+  # a telemetry signal.
+  #
+  # A telemetry provider is used to emit telemetry data. By default, the
+  # {NoOpTelemetryProvider} will not record or emit any telemetry data.
+  # The SDK currently supports OpenTelemetry (OTel) as a provider. See
+  # {OTelProvider} for more information.
+  #
+  # If a provider isn't supported, you can implement your own provider by
+  # inheriting the following base classes and implementing the interfaces
+  # defined:
+  # * {TelemetryProviderBase}
+  # * {ContextManagerBase}
+  # * {TracerProviderBase}
+  # * {TracerBase}
+  # * {SpanBase}
+  module Telemetry
+    class << self
+      # @api private
+      def otel_loaded?
```

Comment:
Is there a reason we need this in the base `Telemetry` module instead of just defining it in the Otel implementation (eg on the `OTelProvider` class)?